teacher.rb

```diff
@@ -0,0 +1,12 @@
+require './person'
+
+class Teacher < Person
+  def initialize(specialization, age, name = 'Unknown')
+    super(age, name)
+    @specialization = specialization
+  end
```

Comment:
- [ ] Ouch! this is a close one, but since the `Teacher` class inherits from the `Person` class it's important to ensure that all the necessary parameters are included in the constructor. I noticed that the `parent_permission` parameter is missing. so kindly add the parameter at line `4` with the default value set to `true` and also make sure to use it as an argument when calling the parent constructor at line `5`.

Reply:
Thank you @AbelG101. I thought this parameter was not needed for teachers, but now I have included it 🙂

Reply:
Hey @Zilola-Nazarova, yeah I know it's kind of silly, teachers don't usually need parent's permission 😅 but since we're inheriting from the `Person` class we'll need to include all attributes from the parent class to properly call the constructor using `super()`